internal/kv/json_file.go

```diff
@@ -0,0 +1,104 @@
+package kv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/brexhq/substation/internal/file"
+)
+
+// kvJSONFile is a read-only key-value store that is derived from a file containing
+// an object and stored in memory.
+//
+// For example, if the file contains this data:
+//
+//	{"foo":"bar","baz":"qux","quux":"corge"}
+//
+// The store becomes this:
+//
+//	map[foo:bar baz:qux quux:corge]
+type kvJSONFile struct {
+	// File contains the location of the text file. This can be either a path on local
+	// disk, an HTTP(S) URL, or an AWS S3 URL.
+	File  string `json:"file"`
+	mu    sync.Mutex
+	items map[string]interface{}
+}
+
+func (store *kvJSONFile) String() string {
+	return toString(store)
+}
+
+// Get retrieves a value from the store.
```

Comment:
Should the get semantics for the JSON support more rich selections using `gjson` rather than a single level of object key lookups? This would let the processor support more complicated datatypes (like when an object has keys or arrays that are desired) and be more consistent with the rest of Substation.

Reply:
does that differ from this? https://github.com/brexhq/substation/blob/jshlbrd/caching/process/kv_store.go#L105-L120

Reply:
I think so. Looks like this implements a gjson type selection for the keys we get/set from the data we are processing but not across the data we are doing the lookup against.

Reply:
done -- the JSON file content is now bytes that is parsed using `internal/json`